### PR TITLE
nspr: patch out timestamp-based impurities

### PR DESCRIPTION
**Pull request by Ekleog**
###### Motivation for this change

https://r13y.com/diff/9c582ff3a3e9892dc2ae5d017110cfbac0f7d83d8614e4e9ff436a5467d46013-54f2851d51fb8de80f54619cd6249fc00f93119d092b34dab5d7a48d2ccfce4c.html

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

@GrahamcOfBorg build nspr 

cc @grahamc for the reproducibility project, as this package appears to have no maintainer nor regular committer

### CODE REVIEW

File: pkgs/development/libraries/nspr/default.nix
Comment (flokli) [inline pkgs/development/libraries/nspr/default.nix]: This will silently fail when it doesn't apply anymore. 

Could we instead craft a patch that uses [SOURCE_DATE_EPOCH](https://reproducible-builds.org/docs/source-date-epoch/)?

This should probably something upstream would be interested in, too.

Comment (Ekleog) [reply pkgs/development/libraries/nspr/default.nix]: Indeed! I've changed the patch and sent it upstream, though it's still waiting for moderator approval over there.

**Review by andir — APPROVED** [review]
I was able to reproduce the same build multiple times with `--check` after the initial build.

It also evals just fine here.

**Comment by andir** [comment]
@GrahamcOfBorg eval
@GrahamcOfBorg build nspr

**Comment by andir** [comment]
I think we can close this as it was already done in 885d1dfe6011885d43b556a4e307a45da97aa8a4.

**Comment by Ekleog** [comment]
Oh right, hadn't noticed, thanks!

**Pull request closed without merging**
